README.md

```diff
@@ -1,2 +1,125 @@
+
+### runForSync
+
+```void runForSync(Function<SELF, Function<Long, Object>> func)```
+
+This method intends for execution synchronous tasks. It receives a function, that should return the execution to be tested for the given the state. For instance:
+
+```
+public static void main(String[] args) {
+    BenchmarksSettings settings = ...;
+    new RouterBenchmarksState(settings).runForSync(state -> {
+
+      Timer timer = state.timer("timer");
+      Router router = state.getRouter();
+
+      return iteration -> {
+        Timer.Context timeContext = timer.time();
+        ServiceReference serviceReference = router.route(request);
+        timeContext.stop();
+        return serviceReference;
+      };
+    });
+  }
+```
+
+As you see, to use this method you need return some function, to generate one you can use the transferred state and iteration's number.
+
+### runForAsync
+
+```void runForAsync(Function<SELF, Function<Long, Publisher<?>>> func)```
+
+This method intends for execution asynchronous tasks. It receives a function, that should return the execution to be tested for the given the state. Note, the unitOfwork should return some `Publisher`. For instance:
+
+```
+  public static void main(String[] args) {
+    BenchmarksSettings settings = ...;
+    new ExampleServiceBenchmarksState(settings).runForAsync(state -> {
+
+      ExampleService service = state.exampleService();
+      Meter meter = state.meter("meter");
+
+      return iteration -> {
+        return service.invoke("hello")
+            .doOnTerminate(() -> meter.mark());
+      };
+    });
+  }
+```
+
+As you see, to use this method you need return some function, to generate one you can use the transferred state and iteration's number.
+
+### runWithRampUp
+
+```
+<T> void runWithRampUp(BiFunction<Long, SELF, Publisher<T>> setUp,
+                       Function<SELF, BiFunction<Long, T, Publisher<?>>> func,
+                       BiFunction<SELF, T, Mono<Void>> cleanUp)
+```
+
+This method intends for execution asynchronous tasks with consumption some resources via ramp-up strategy. It receives three functions, they are necessary to provide all resource life-cycle. The first function is like resource supplier, to implement this one you have access to the active state and a ramp-up iteration's number. And when ramp-up strategy asks for new resources it will be invoked. The second function is like unitOfWork supplier, to implement this one you receive the active state (to take some services or metric's tools), iteration's number and a resource, that was created on the former step. And the last function is like clean-up supplier, that knows how to need release given resource. For instance:
+
+```
+  public static void main(String[] args) {
+    BenchmarksSettings settings = ...;
```

Comment:
Better to put real ramp up settings right here, and in description below, you could refer to them  so the reader would understand what this ramp-up thing is all about. So far text is too verbose and needs more factual content.